<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom">

  <title><![CDATA[Category: python, | Ещё один блог]]></title>
  <link href="http://Anshik.github.com/blog/categories/python-/atom.xml" rel="self"/>
  <link href="http://Anshik.github.com/"/>
  <updated>2012-11-16T15:27:29+04:00</updated>
  <id>http://Anshik.github.com/</id>
  <author>
    <name><![CDATA[Anshik]]></name>
    
  </author>
  <generator uri="http://octopress.org/">Octopress</generator>

  
  <entry>
    <title type="html"><![CDATA[Отладка почтовых сообщений в django]]></title>
    <link href="http://Anshik.github.com/blog/2012/05/29/otladka-pochtovykh-soobshchienii-v-django/"/>
    <updated>2012-05-29T15:42:00+04:00</updated>
    <id>http://Anshik.github.com/blog/2012/05/29/otladka-pochtovykh-soobshchienii-v-django</id>
    <content type="html"><![CDATA[<p>python -m smtpd -n -c DebuggingServer localhost:25
Эта одностроковая команда запускает сервер SMTP на порту 1025. Здесь используется стандартная библиотека smtpd для языка Python (задается при помощи -m smtpd) и передаются три аргумента: -n, -c DebuggingServer и localhost:1025.</p>

<!-- more -->


<p>Аргумент -n указывает Python-у не устанавливать для setuid (позволяет изменять пользователя) значение "nobody" — код будет работать под вашим пользователем.
Аргумент -c DebuggingServer указывает Python-у использовать класс DebuggingServer в качестве реализации SMTP, которая печатает каждое получаемое сообщение в выходной поток stdout.
Аргумент localhost:1025 указывает Python-у запускать SMTP сервер как locahost на порту1025.
 Однако, если вы захотите запустить сервер на стандартном порту 25, вам нужно будет использовать команду sudo, поскольку только пользователь с правами root может запускать сервисы на портах 1 — 1024. Они также известны как привилегированные порты.</p>

<p>Есть ещё какие-то приложения, которые запускаешь, шлёшь туда отладочную почту.</p>
]]></content>
  </entry>
  
  <entry>
    <title type="html"><![CDATA[ipython for python3.2 WindowsXP]]></title>
    <link href="http://Anshik.github.com/blog/2012/05/29/ipython-for-python3-dot-2-windowsxp/"/>
    <updated>2012-05-29T15:19:00+04:00</updated>
    <id>http://Anshik.github.com/blog/2012/05/29/ipython-for-python3-dot-2-windowsxp</id>
    <content type="html"><![CDATA[<p>Вот, задался целью всё-таки забороть ipython для Python 3.2, и получилось :)</p>

<h3>Делал примерно так:</h3>

<p>1.1. Поставил Python3.2: python-3.2.1.msi<br/>
1.2. Пропишите пути к новому питону</p>

<ol>
<li>distribute_setup.py : http://python-distribute.org/distribute_setup.py<br/>
это для того чтобы заработал easy_install</li>
</ol>


<!-- more -->


<ol>
<li><p>Дальше<br/>
<code>easy_install http://archive.ipython.org/release/0.11/py3/ipython-0.11-py3.2.egg</code><br/>
ставится ipython к папке <code>c:/python32/Scripts</code> тоже хорошо бы переменную <strong>path</strong> виндовса поправить</p></li>
<li><p>Запускаем <code>ipython3.exe</code> в папке Scripts и ловим эксепшн, у меня во всяком случае было так:</p></li>
</ol>


<p>``` bash
 c:\Python32\Scripts>ipython3.exe
 Error in sys.excepthook:
 TypeError: _isdir() takes exactly 1 argument (0 given)</p>

<p> Original exception was:
 Traceback (most recent call last):
 File "c:\Python32\Scripts\ipython3-script.py", line 9, in
 load_entry_point('ipython==0.11', 'console_scripts', 'ipython3')()
 File "c:\Python32\lib\site-packages\ipython-0.11-py3.2.egg\IPython\frontend\te
rminal\ipapp.py", line 369, in launch_new_instance
 app.initialize()
 File "c:\Python32\lib\site-packages\ipython-0.11-py3.2.egg\IPython\frontend\te
rminal\ipapp.py", line 283, in initialize
 self.init_shell()
 File "c:\Python32\lib\site-packages\ipython-0.11-py3.2.egg\IPython\frontend\te
rminal\ipapp.py", line 303, in init_shell
 ipython_dir=self.ipython_dir)
 File "c:\Python32\lib\site-packages\ipython-0.11-py3.2.egg\IPython\config\conf
igurable.py", line 295, in instance
 inst = cls(*args, **kwargs)
 File "c:\Python32\lib\site-packages\ipython-0.11-py3.2.egg\IPython\frontend\te
rminal\interactiveshell.py", line 112, in <strong>init</strong>
 user_global_ns=user_global_ns, custom_exceptions=custom_exceptions
 File "c:\Python32\lib\site-packages\ipython-0.11-py3.2.egg\IPython\core\intera
ctiveshell.py", line 384, in <strong>init</strong>
 self.db = PickleShareDB(os.path.join(self.profile_dir.location, 'db'))
 File "c:\Python32\lib\site-packages\ipython-0.11-py3.2.egg\IPython\utils\pickl
eshare.py", line 52, in <strong>init</strong>
 if not self.root.isdir():
TypeError: _isdir() takes exactly 1 argument (0 given)</p>

<p>c:\Python32\Scripts>
```</p>

<p>Решил я это так
В файле <code>"c:\Python32\lib\site-packages\ipython-0.11-py3.2.egg\IPython\utils\pickleshare.py"</code><br/>
В 52 строке закомментировал<br/>
<code>#if not self.root.isdir():</code><br/>
И 53-ей строкой написал следующее<br/>
<code>if not os.path.isdir(self.root):</code></p>

<p>Заработало, что не может не радовать.
Единственное pyreadline ещё похоже нужно замучить</p>
]]></content>
  </entry>
  
</feed>
